bigchaindb/web/websocket_server.py

```diff
@@ -0,0 +1,174 @@
+"""WebSocket server for the BigchainDB Event Stream API."""
+
+# NOTE
+#
+# This module contains some functions and utilities that might belong to other
+# modules. For now, I prefer to keep everything in this module. Why? Because
+# those functions are needed only here.
+#
+# When we will extend this part of the project and we find that we need those
+# functionalities elsewhere, we can start creating new modules and organizing
+# things in a better way.
+
+
+import json
+import asyncio
+import logging
+import threading
+from uuid import uuid4
+
+import aiohttp
+from aiohttp import web
+
+from bigchaindb import config
+from bigchaindb.events import EventTypes
+
+
+logger = logging.getLogger(__name__)
+POISON_PILL = 'POISON_PILL'
+EVENTS_ENDPOINT = '/api/v1/streams/valid_tx'
+
+
+def _put_into_capped_queue(queue, value):
+    """Put a new item in a capped queue.
+
+    If the queue reached its limit, get the first element
```

Comment:
I think since we don't have the problem of high throughput right now, it might be safer to make it a blocking queue, ie, doesn't drop messages. If a client is not reading fast enough, then the server may get blocked waiting for their socket to become writeable. In that case, the slow client socket should be somehow dealt with. If it isn't dealt with, a slow client may interfere with what other clients see just by connecting and reading slowly.